### PR TITLE
build: add pip dependencies

### LINKED ISSUE #208: fix CI builds
vis. https://github.com/TomographicImaging/iDVC/actions/runs/6930225979 (#204)

### LINKED ISSUE #215: fix pip install
`setup.py` is missing dependencies. That means dev install currently requires:

```sh
wget https://raw.githubusercontent.com/TomographicImaging/iDVC/master/recipe/dev_environment.yml
sed -i '/.*ccpi-dvc.*/d' dev_environment.yml
conda env create -n idvc -f dev_environment.yml
conda activate idvc
pip install "git+https://github.com/TomographicImaging/iDVC@master#egg=idvc"
```

### PR DESCRIPTION
**Pull request by casperdcl**
- fix (some) missing pip dependencies (fixes #215)
- update workflows & fix tests (fixes #208)

### CODE REVIEW

**Review by paskino — COMMENTED** [review]
The cilviewer is a dependency but it is not available in pypi and the build fails. How do we proceed?

**Comment by casperdcl** [comment]
I added an ultra basic test (actually try to import something!) and was missing `libgl-dev` to `import vtk`.

ready to merge now :)

File: src/idvc/__init__.py [hunk 1]
```diff
@@ -12,3 +12,7 @@
 
 #   Author: Laura Murgatroyd (UKRI-STFC)
 #   Author: Edoardo Pasca (UKRI-STFC)
+try:
+    import ccpi.viewer
```
Comment (DanicaSTFC) [inline src/idvc/__init__.py:16]: The viewer has been moved to https://github.com/TomographicImaging/CILViewer. Can we update this?

**Review by DanicaSTFC — CHANGES_REQUESTED** [review]

**Review by paskino — APPROVED** [review]

File: src/idvc/__init__.py (hunk 1 above)
Comment (DanicaSTFC) [reply src/idvc/__init__.py:16]: Apparently the conda is still called ccpi so my comment is not relevant.

**Review by DanicaSTFC — APPROVED** [review]

**Pull request merged**